code/game/machinery/race_converter.dm

```diff
@@ -0,0 +1,149 @@
+	var/iterations = 0 // how long the user (victim) has been in the chamber for
+	var/changed =  FALSE
+	var/datum/species/desired_race = /datum/species/human/felinid
+	var/datum/looping_sound/microwave/soundloop
+
+/obj/machinery/species_converter/racewar
+	name = "species hypnosis chamber"
+	brainwash = TRUE
+
+/obj/machinery/species_converter/Initialize()
+	. = ..()
+	soundloop = new(list(src),  FALSE)
+	update_icon()
+
+/obj/machinery/species_converter/Destroy()
+	QDEL_NULL(soundloop)
+	return ..()
+
+/obj/machinery/species_converter/close_machine(mob/user)
+	if(panel_open)
+		to_chat(user, "<span class='warning'>You need to close the maintenance hatch first!</span>")
+		return
+	..()
+	playsound(src, 'sound/machines/click.ogg', 50)
+	if(occupant)
+		if(!iscarbon(occupant))
+			occupant.forceMove(drop_location())
+			occupant = null
+			return
+		to_chat(occupant, "<span class='notice'>You enter [src]</span>")
+		addtimer(CALLBACK(src, .proc/start_extracting), 20, TIMER_OVERRIDE|TIMER_UNIQUE)
+		update_icon()
+
+/obj/machinery/species_converter/open_machine(mob/user)
+	iterations = 0
+	playsound(src, 'sound/machines/click.ogg', 50)
+	if(processing)
+		stop()
+	..()
+
+/obj/machinery/species_converter/proc/stop()
+	processing = FALSE
+	soundloop.stop()
+	set_light(0, 0)
+
+/obj/machinery/species_converter/interact(mob/user)
+	if(state_open)
+		close_machine()
+	else if(!processing)
+		open_machine()
+	else
+		to_chat(user, "<span class='warning'>You can't open the [src] while it's active!</span>")
+
+/obj/machinery/species_converter/update_overlays()
+	. = ..()
+	if(!state_open)
+		if(processing)
+			. += "[icon_state]_door_on"
+			. += "[icon_state]_stack"
+			. += "[icon_state]_smoke"
+			. += "[icon_state]_green"
+		else
+			. += "[icon_state]_door_off"
+			if(occupant)
+				if(powered(AREA_USAGE_EQUIP))
+					. += "[icon_state]_stack"
+					. += "[icon_state]_yellow"
+			else
+				. += "[icon_state]_red"
+	else if(powered(AREA_USAGE_EQUIP))
+		. += "[icon_state]_red"
+	if(panel_open)
+		. += "[icon_state]_panel"
+
+/obj/machinery/species_converter/process()
+	if(!processing)
+		return
+	if(!is_operational() || !occupant || !iscarbon(occupant))
+		open_machine()
+		return
+
+	var/mob/living/carbon/C = occupant
+	if(is_species(C, desired_race))
+		open_machine()
+		playsound(src, 'sound/machines/microwave/microwave-end.ogg', 100, FALSE)
+		return
+
+	else if(prob(iterations * 10 + 10)) // conversion has some random variation in it
+		C.set_species(desired_race)
+		if(brainwash)
+			to_chat(C, "<span class='userdanger'>A new compulsion fills your mind... you feel forced to obey it!</span>")
+			var/objective = "Convert as many people as possible into a [initial(desired_race.name)]. Racewar!"
+			brainwash(C, objective)
+			log_game("[key_name(C)] has been brainwashed with the objective '[objective]' via the species converter.")
+
+	iterations++
+	use_power(500)
+
+/obj/machinery/species_converter/proc/start_extracting()
```

Comment:
I do think we need a better name than copypasta of `/obj/machinery/fat_sucker/proc/start_extracting()`

Reply:
Forgot to change, oops